File: Tutorial/Examples/Topic_routing/Producer_Python/Producer.py
import pika
import random
from pika.exchange_type import ExchangeType


# Set connection parameters. If connecting to a real server localhost should be replaced by the server´s address.
connection_parameters = pika.ConnectionParameters('localhost')

# Instantiate a connection unsing the connection_parameters previously defined
connection = pika.BlockingConnection(connection_parameters)

# Instantiate a channel
channel = connection.channel()

# Instantiate an exchange
channel.exchange_declare(exchange='topic', exchange_type=ExchangeType.topic)

# Define the message to send
l = random.randint(1, 6)
message = f"{l}"

# Select the routing key to use when sending the message.
routing_keys = ['square.perimeter', 'square.area']
key = routing_keys[random.randint(0, len(routing_keys)-1)]

# Publish the message to the letterbox queue
channel.basic_publish(exchange='topic', routing_key=key, body=message)

# Print the sent message
print(f'Sent message: {message} with the key {key}')

# Close connection
connection.close()

File: Tutorial/Examples/Topic_routing/Producer_C#/Producer.cs
using System;
using System.Text;
using RabbitMQ.Client;

class Program{
    static void Main(string[] args){
        //Variable declaration.
        int l;

        // Create connection.
        var factory = new ConnectionFactory{ HostName = "localhost"};

        // Instantiate the connection and one channel.
        using var connection = factory.CreateConnection();
        using var channel = connection.CreateModel();

        // Declare an exchange called routing.
        channel.ExchangeDeclare(exchange: "topic", type: ExchangeType.Topic);

        // Instantiate a random number generator
        Random random = new Random();


        try{
            l = int.Parse(args[0]);
        }
        catch{
            l = random.Next(1, 6);
            Console.WriteLine("Invalid arguments. Random value asigned");

        }

        // Declare message to produce and encode it as bytes
        var message = string.Format("{0}", l);
        var encodedMessage = Encoding.UTF8.GetBytes(message);

        // Select key to use for topic the message.
        string[] keys = {"square.area", "square.perimeter"}; 
        string key = keys[random.Next(0, keys.Length)];

        // Start publishing to the exchange.
        channel.BasicPublish(exchange: "topic", routingKey: key, null, encodedMessage);

        // Print the published message to console.
        Console.WriteLine($"Sent message: {message} with the key {key}");
    }
}